libraries/chain/database.cpp

```diff
@@ -3746,18 +3746,18 @@ void database::clear_expired_delegations()
 }
 #ifdef STEEM_ENABLE_SMT
 template< typename smt_balance_object_type, class balance_operator_type >
-void database::adjust_smt_balance( const account_name_type& name, const asset& delta, bool check_account,
-   balance_operator_type balance_operator )
+void database::adjust_smt_balance( const account_name_type& name, const asset_symbol_type& delta_symbol, bool positive_delta, bool zero_delta,
```

Comment:
If we are going to use a templated generic method with specific implementation in a class type, let's also encode such fields in the class. Both `smt_regular_balance_operator` and `smt_reward_balance_operator` contains assets, which contain the asset symbols. And as far as I can tell, `positive_delta` and `zero_delta` can either be calculated from the `balance_operator_type` or can be encoded if needed.

Also, the `balance_operator` should be a reference, not a value type.

Reply:
Done. See commit https://github.com/steemit/steem/pull/2356/commits/a897d4a2fa11cc33a6d79040f934574a7125af0d